[Zhang Leran] iP

zlllllr opened the pull request:
### **Orim**

> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

Orim frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/128).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing task

- [ ] Managing deadlines (coming soon)

- [ ] Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
        new Orim("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zlllllr committed:
Rename main, add skeletal version of greet and bye

zlllllr committed:
Add support to echo

zlllllr committed:
add support to store

zlllllr committed:
add support to mark and unmark; store task as object

zlllllr committed:
add,todo deadline event class

zlllllr committed:
add todo etc to main

zlllllr committed:
add support to handle error

zlllllr committed:
add support to delete task

zlllllr committed:
Add support to store history when exit

zlllllr committed:
Add support to load history when start the bot

zlllllr committed:
Merge branch 'branch-Level-7'

zlllllr committed:
Allow the bot to understand date

zlllllr committed:
Merge branch 'branch-Level-8'

zlllllr committed:
Add packages

zlllllr committed:
Put all classes in packages

zlllllr committed:
Refactor the code into different classes and put all classes in packages

zlllllr committed:
Merge branch 'add-gradle-support'

zlllllr committed:
Remove redundant import in Orim.java

zlllllr committed:
Set mainclass in build.gradle to correct location

zlllllr committed:
Add test for Parser.parse() and Todo.generateLog()

zlllllr committed:
Change archiveBaseName in build.gradle

zlllllr committed:
Add JavaDoc to public class and method

zlllllr committed:
Tweak the code to follow coding standard

zlllllr committed:
Add support to find a task according to its description.

zlllllr committed:
Merge branch 'branch-A-JavaDoc'

zlllllr committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/orim/task/Task.java

zlllllr committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/orim/command/DisplayCommand.java

Rae2480 commented on src/main/java/orim/Orim.java:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Represents a chatbot that allows user to keep track of their tasks.
+ */
+public class Orim {
```
whoa! i like how clean this file is! no unnecessary methods or anything... Good job!

Rae2480 commented on src/main/java/orim/command/ExitCommand.java:
```diff
@@ -0,0 +1,34 @@
+        super(true);
+    }
+
+    /**
+     * @param tasks TaskList that contain all the tasks.
+     * @param ui deals with interactions with the user.
+     * @param file deals with loading tasks from the file and saving tasks in the file.
+     * @throws RuntimeException if file cannot be open with FileWriter.
+     */
```
your javadoc comments seem quite clear, which is nice. maybe you could add 1-2 lines at the start as a summary?

Rae2480 commented on src/main/java/orim/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+        return "D," + (isDone ? 1 : 0) + "," + super.description + "," + time;
+    }
+
+    public String toString() {
```
nit: all public methods should have javadoc headers. but i understand this wasnt in the iP requirements so i wouldn't worry about it too much!

Rae2480 commented on src/main/java/orim/ui/Ui.java:
```diff
@@ -0,0 +1,45 @@
+package orim.ui;
+
+import orim.tasklist.TaskList;
+
+import java.util.Scanner;
```
your import statements are so well-organised! nice work :D

Rae2480 commented on src/main/java/orim/command/UpdateCommand.java:
```diff
@@ -0,0 +1,36 @@
+public class UpdateCommand extends Command {
+
+    private final int index;
+    private final boolean shouldMark;
```
i understard when shouldMark = true then task is marked as done but the inverse is less easily understood. shouldMark = false isnt as clear, i mean. Maybe you can consider renaming this boolean variable to isMarkCommand?

Rae2480 reviewed (commented):
LGTM! Your code is pretty error-free and nothing really stood out as glaringly bad. The only issue is that shouldMark boolean variable and everything else is clear, logical and clean. All the best with future iP increments :D

PriyadarshiCharvi commented on src/main/java/orim/Orim.java:
```diff
@@ -0,0 +1,54 @@
+package orim;
```
LGTM! The file looks very clean and readable. Good job!

PriyadarshiCharvi commented on src/main/java/orim/command/AddCommand.java:
```diff
@@ -0,0 +1,31 @@
+        this.task = task;
+    }
+
+    /**
+     * @param tasks TaskList that contain all the tasks.
+     * @param ui deals with interactions with the user.
+     * @param file deals with loading tasks from the file and saving tasks in the file.
+     */
```
A good rule to follow for javadocs is to add a short summary of what the method does as the first line. Could you try to incorporate that?

PriyadarshiCharvi commented on src/main/java/orim/command/SearchCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+    private final String searchKey;
+
+    public SearchCommand(String searchKey) {
+
+        super(false);
+        this.searchKey = searchKey;
+
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage file) {
+        TaskList ans = tasks.find(searchKey);
+        Ui.showListing(ans);
+    }
```
Do you think you could add header comments for all public methods such as these? It would make the code more readable and uniform!

PriyadarshiCharvi commented on src/main/java/orim/parser/Parser.java:
```diff
@@ -0,0 +1,92 @@
+        List<String> strings = Arrays.stream(command.split("/"))
+                .map(x -> x.trim())
+                .toList();
+        switch (strings.get(0).split(" ")[0]) {
+        case "todo":
+            try {
+                temp = new AddCommand(new Todo(strings.get(0).substring(5)));
+            } catch (IndexOutOfBoundsException e) {
+                throw new MissingContentException(" OOPS!!! The description of a todo cannot be empty.");
+            }
+            break;
+        case "deadline":
+            try {
+                temp = new AddCommand(new Deadline(strings.get(0).substring(9)
+                        , strings.get(1).substring(3)));
+            } catch (IndexOutOfBoundsException e) {
+                throw new MissingContentException(" OOPS!!! The description of a deadline cannot be empty.");
+            }
+            break;
+        case "event":
+            try {
+                temp = new AddCommand(new Event(strings.get(0).substring(6),
```
I like the use of switch-case statement here! Makes it easy to understand, great work!

PriyadarshiCharvi commented on src/test/java/orim/parser/ParserTest.java:
```diff
@@ -0,0 +1,22 @@
+package orim.parser;
+
+import org.junit.jupiter.api.Test;
+import orim.myexception.IncorrectInputException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
```
I like your organisation of import statements! Looks very clean and consistent

PriyadarshiCharvi commented on src/test/java/orim/parser/ParserTest.java:
```diff
@@ -0,0 +1,22 @@
+
+public class ParserTest {
+    @Test
+    public void parsar_incorrectInput_exceptionThrown() {
```
Nothing entirely wrong with this line per se, I'd just suggest correcting the spelling from 'parsar' to 'parser'!

PriyadarshiCharvi reviewed (commented):
Overall, your code looks really clean and well organised. Would just suggest adding short summaries of methods as the first line of javadocs. Otherwise, great job!

zlllllr committed:
Add GUI to the program

zlllllr committed:
Update gui package

zlllllr committed:
Merge branch 'branch-Level-10'

zlllllr committed:
Add assertion in task creation to ensure no empty description

zlllllr committed:
Add javadoc for SearchCommand class

zlllllr committed:
Merge pull request #2 from zlllllr/branch-A-Assertions

Add assertion in task creation to ensure no empty description

zlllllr committed:
Merge branch 'master' into branch-A-CodeQuality

zlllllr committed:
Merge pull request #3 from zlllllr/branch-A-CodeQuality

Branch a code quality

zlllllr committed:
SearchCommand: add advancedSearch method

SearchCommand should allow users to search based on special needs, such as find unmarked tasks

Let's,
* add one attribute to SearchCommand class to detect whether the search command is advanced
* add findUnmarkTasks in TaskList to find all unmark tasks in the tasklist

zlllllr committed:
Add user guide and sample Ui

zlllllr committed:
Fix bugs

zlllllr committed:
Fix bugs

zlllllr committed:
Remove printing output to the terminal

zlllllr committed:
Update README.md

zlllllr committed:
Add picture to README.md
